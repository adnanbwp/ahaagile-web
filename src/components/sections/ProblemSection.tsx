import React from 'react';
import { Card, CardContent } from '@/components/ui/card';
import { Clock, DollarSign, AlertTriangle, Frown } from 'lucide-react';

interface ProblemSectionProps {
  className?: string;
}

export default function ProblemSection({ className = '' }: ProblemSectionProps): JSX.Element {
  const painPoints = [
    {
      icon: Clock,
      title: 'Time Drain',
      description: 'Manual processes consuming excessive employee hours and reducing productivity'
    },
    {
      icon: DollarSign,
      title: 'Hidden Costs',
      description: 'Invisible expenses from inefficient workflows and repeated manual tasks'
    },
    {
      icon: AlertTriangle,
      title: 'Error Prone',
      description: 'Human mistakes in manual processes leading to costly corrections and delays'
    },
    {
      icon: Frown,
      title: 'Employee Burnout',
      description: 'Repetitive manual work reducing job satisfaction and increasing turnover'
    }
  ];

  return (
    <section className={`section-spacing bg-muted ${className}`}>
      <div className="container-brand">
        {/* Section Header */}
        <div className="text-center mb-16">
          <h2 className="text-hero text-foreground mb-8">
            The Hidden Cost of Manual Workflows
          </h2>
          
          {/* Financial Impact Callout */}
          <div className="inline-block px-8 py-6 rounded-2xl mb-8 bg-[var(--gradient-primary)]">
            <div className="text-center">
              <h3 className="text-4xl md:text-5xl lg:text-6xl font-inter font-bold text-primary-foreground mb-2">
                $236K - $330K
              </h3>
              <p className="text-xl md:text-2xl font-inter text-primary-foreground/80">
                Annual Cost of Inefficiency
              </p>
            </div>
          </div>
        </div>

        {/* Pain Points Grid */}
        <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-6 mb-16">
          {painPoints.map((point, index) => {
            const IconComponent = point.icon;
            return (
              <Card 
                key={index}
                className="card-elevated hover:shadow-xl transition-all duration-300 hover:scale-105"
              >
                <CardContent className="p-6 text-center">
                  <div className="w-16 h-16 mx-auto mb-4 bg-gradient-accent rounded-full flex items-center justify-center">
                    <IconComponent className="w-8 h-8 text-foreground" />
                  </div>
                  <h3 className="text-xl font-inter font-semibold text-foreground mb-3">
                    {point.title}
                  </h3>
                  <p className="text-muted-foreground leading-relaxed">
                    {point.description}
                  </p>
                </CardContent>
              </Card>
            );
          })}
        </div>

        {/* Transition CTA */}
        <div className="text-center">
          <div className="inline-flex items-center space-x-2 text-foreground font-inter font-semibold">
            <span className="text-lg">Ready to reclaim your time and reduce costs?</span>
            <div className="w-0 h-0 border-l-8 border-r-8 border-t-8 border-l-transparent border-r-transparent border-t-foreground animate-bounce"></div>
          </div>
          <p className="text-muted-foreground mt-2">
            Discover how our intelligent automation solutions can transform your workflows below.
          </p>
        </div>
      </div>
    </section>
  );
}