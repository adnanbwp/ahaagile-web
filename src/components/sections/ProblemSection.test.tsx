import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import ProblemSection from './ProblemSection';

// Mock Lucide React icons
jest.mock('lucide-react', () => ({
  Clock: ({ className }: { className: string }) => <div data-testid="clock-icon" className={className} />,
  DollarSign: ({ className }: { className: string }) => <div data-testid="dollar-sign-icon" className={className} />,
  AlertTriangle: ({ className }: { className: string }) => <div data-testid="alert-triangle-icon" className={className} />,
  Frown: ({ className }: { className: string }) => <div data-testid="frown-icon" className={className} />,
}));

describe('ProblemSection', () => {
  beforeEach(() => {
    render(<ProblemSection />);
  });

  describe('Section Header and Financial Impact', () => {
    it('renders the correct section header', () => {
      expect(screen.getByRole('heading', { level: 2, name: 'The Hidden Cost of Manual Workflows' })).toBeInTheDocument();
    });

    it('displays the financial impact callout with correct text', () => {
      expect(screen.getByRole('heading', { level: 3, name: '$236K - $330K' })).toBeInTheDocument();
      expect(screen.getByText('Annual Cost of Inefficiency')).toBeInTheDocument();
    });

    it('applies proper typography classes to the header', () => {
      const header = screen.getByRole('heading', { level: 2, name: 'The Hidden Cost of Manual Workflows' });
      expect(header).toHaveClass('text-hero', 'text-brand-navy');
    });

    it('applies proper typography classes to the financial impact', () => {
      const financialImpact = screen.getByRole('heading', { level: 3, name: '$236K - $330K' });
      expect(financialImpact).toHaveClass('font-inter', 'font-bold', 'text-white');
    });
  });

  describe('Pain Points Cards', () => {
    it('renders all four pain point cards', () => {
      expect(screen.getByText('Time Drain')).toBeInTheDocument();
      expect(screen.getByText('Hidden Costs')).toBeInTheDocument();
      expect(screen.getByText('Error Prone')).toBeInTheDocument();
      expect(screen.getByText('Employee Burnout')).toBeInTheDocument();
    });

    it('displays correct descriptions for each pain point', () => {
      expect(screen.getByText('Manual processes consuming excessive employee hours and reducing productivity')).toBeInTheDocument();
      expect(screen.getByText('Invisible expenses from inefficient workflows and repeated manual tasks')).toBeInTheDocument();
      expect(screen.getByText('Human mistakes in manual processes leading to costly corrections and delays')).toBeInTheDocument();
      expect(screen.getByText('Repetitive manual work reducing job satisfaction and increasing turnover')).toBeInTheDocument();
    });

    it('renders appropriate icons for each pain point', () => {
      expect(screen.getByTestId('clock-icon')).toBeInTheDocument();
      expect(screen.getByTestId('dollar-sign-icon')).toBeInTheDocument();
      expect(screen.getByTestId('alert-triangle-icon')).toBeInTheDocument();
      expect(screen.getByTestId('frown-icon')).toBeInTheDocument();
    });

    it('applies hover effects classes to cards', () => {
      const cards = document.querySelectorAll('.card-elevated');
      expect(cards).toHaveLength(4);
      cards.forEach(card => {
        expect(card).toHaveClass('hover:shadow-xl', 'transition-all', 'duration-300', 'hover:scale-105');
      });
    });
  });

  describe('Transition CTA', () => {
    it('renders the transition CTA text', () => {
      expect(screen.getByText('Ready to reclaim your time and reduce costs?')).toBeInTheDocument();
      expect(screen.getByText('Discover how our intelligent automation solutions can transform your workflows below.')).toBeInTheDocument();
    });

    it('applies proper styling to transition CTA', () => {
      const ctaText = screen.getByText('Ready to reclaim your time and reduce costs?');
      expect(ctaText).toHaveClass('text-lg');
      expect(ctaText.parentElement).toHaveClass('text-brand-navy', 'font-inter', 'font-semibold');
    });
  });

  describe('Responsive Design', () => {
    it('applies responsive grid classes', () => {
      const grid = document.querySelector('.grid');
      expect(grid).toBeInTheDocument();
      expect(grid).toHaveClass('grid-cols-1', 'md:grid-cols-2', 'lg:grid-cols-4');
    });

    it('applies responsive typography to financial impact', () => {
      const financialImpact = screen.getByRole('heading', { level: 3, name: '$236K - $330K' });
      expect(financialImpact).toHaveClass('text-4xl', 'md:text-5xl', 'lg:text-6xl');
    });
  });

  describe('Accessibility', () => {
    it('uses proper semantic HTML structure', () => {
      const section = document.querySelector('section');
      expect(section).toBeInTheDocument();
      expect(screen.getAllByRole('heading')).toHaveLength(6); // h2 + h3 + 4 card titles
    });

    it('provides meaningful content for screen readers', () => {
      const section = document.querySelector('section');
      expect(section).toBeInTheDocument();
      
      // Check that all text content is accessible
      expect(screen.getByText('The Hidden Cost of Manual Workflows')).toBeInTheDocument();
      expect(screen.getByText('$236K - $330K')).toBeInTheDocument();
      expect(screen.getByText('Annual Cost of Inefficiency')).toBeInTheDocument();
    });
  });

  describe('Custom Props', () => {
    it('applies custom className when provided', () => {
      const { container } = render(<ProblemSection className="custom-class" />);
      const section = container.querySelector('section');
      expect(section).toHaveClass('custom-class');
    });

    it('applies default background and spacing classes', () => {
      const section = document.querySelector('section');
      expect(section).toHaveClass('section-spacing', 'bg-secondary-50');
    });
  });
});

// Snapshot test to prevent visual regression
describe('ProblemSection Snapshots', () => {
  it('matches snapshot for visual regression prevention', () => {
    const { container } = render(<ProblemSection />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot with custom className', () => {
    const { container } = render(<ProblemSection className="test-class" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});