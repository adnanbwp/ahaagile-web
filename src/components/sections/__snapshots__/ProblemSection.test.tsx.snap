// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProblemSection Snapshots matches snapshot for visual regression prevention 1`] = `
<section
  class="section-spacing bg-secondary-50 "
>
  <div
    class="container-brand"
  >
    <div
      class="text-center mb-16"
    >
      <h2
        class="text-hero text-brand-navy mb-8"
      >
        The Hidden Cost of Manual Workflows
      </h2>
      <div
        class="inline-block px-8 py-6 rounded-2xl mb-8"
        style="background: linear-gradient(135deg, var(--brand-navy) 0%, var(--brand-navy-light) 100%);"
      >
        <div
          class="text-center"
        >
          <h3
            class="text-4xl md:text-5xl lg:text-6xl font-inter font-bold text-white mb-2"
          >
            $236K - $330K
          </h3>
          <p
            class="text-xl md:text-2xl font-inter text-gray-200"
          >
            Annual Cost of Inefficiency
          </p>
        </div>
      </div>
    </div>
    <div
      class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-6 mb-16"
    >
      <div
        class="rounded-lg border bg-card text-card-foreground shadow-sm card-elevated hover:shadow-xl transition-all duration-300 hover:scale-105"
      >
        <div
          class="p-6 text-center"
        >
          <div
            class="w-16 h-16 mx-auto mb-4 bg-gradient-accent rounded-full flex items-center justify-center"
          >
            <div
              class="w-8 h-8 text-brand-navy"
              data-testid="clock-icon"
            />
          </div>
          <h3
            class="text-xl font-inter font-semibold text-brand-navy mb-3"
          >
            Time Drain
          </h3>
          <p
            class="text-secondary-600 leading-relaxed"
          >
            Manual processes consuming excessive employee hours and reducing productivity
          </p>
        </div>
      </div>
      <div
        class="rounded-lg border bg-card text-card-foreground shadow-sm card-elevated hover:shadow-xl transition-all duration-300 hover:scale-105"
      >
        <div
          class="p-6 text-center"
        >
          <div
            class="w-16 h-16 mx-auto mb-4 bg-gradient-accent rounded-full flex items-center justify-center"
          >
            <div
              class="w-8 h-8 text-brand-navy"
              data-testid="dollar-sign-icon"
            />
          </div>
          <h3
            class="text-xl font-inter font-semibold text-brand-navy mb-3"
          >
            Hidden Costs
          </h3>
          <p
            class="text-secondary-600 leading-relaxed"
          >
            Invisible expenses from inefficient workflows and repeated manual tasks
          </p>
        </div>
      </div>
      <div
        class="rounded-lg border bg-card text-card-foreground shadow-sm card-elevated hover:shadow-xl transition-all duration-300 hover:scale-105"
      >
        <div
          class="p-6 text-center"
        >
          <div
            class="w-16 h-16 mx-auto mb-4 bg-gradient-accent rounded-full flex items-center justify-center"
          >
            <div
              class="w-8 h-8 text-brand-navy"
              data-testid="alert-triangle-icon"
            />
          </div>
          <h3
            class="text-xl font-inter font-semibold text-brand-navy mb-3"
          >
            Error Prone
          </h3>
          <p
            class="text-secondary-600 leading-relaxed"
          >
            Human mistakes in manual processes leading to costly corrections and delays
          </p>
        </div>
      </div>
      <div
        class="rounded-lg border bg-card text-card-foreground shadow-sm card-elevated hover:shadow-xl transition-all duration-300 hover:scale-105"
      >
        <div
          class="p-6 text-center"
        >
          <div
            class="w-16 h-16 mx-auto mb-4 bg-gradient-accent rounded-full flex items-center justify-center"
          >
            <div
              class="w-8 h-8 text-brand-navy"
              data-testid="frown-icon"
            />
          </div>
          <h3
            class="text-xl font-inter font-semibold text-brand-navy mb-3"
          >
            Employee Burnout
          </h3>
          <p
            class="text-secondary-600 leading-relaxed"
          >
            Repetitive manual work reducing job satisfaction and increasing turnover
          </p>
        </div>
      </div>
    </div>
    <div
      class="text-center"
    >
      <div
        class="inline-flex items-center space-x-2 text-brand-navy font-inter font-semibold"
      >
        <span
          class="text-lg"
        >
          Ready to reclaim your time and reduce costs?
        </span>
        <div
          class="w-0 h-0 border-l-8 border-r-8 border-t-8 border-l-transparent border-r-transparent border-t-brand-navy animate-bounce"
        />
      </div>
      <p
        class="text-secondary-600 mt-2"
      >
        Discover how our intelligent automation solutions can transform your workflows below.
      </p>
    </div>
  </div>
</section>
`;

exports[`ProblemSection Snapshots matches snapshot with custom className 1`] = `
<section
  class="section-spacing bg-secondary-50 test-class"
>
  <div
    class="container-brand"
  >
    <div
      class="text-center mb-16"
    >
      <h2
        class="text-hero text-brand-navy mb-8"
      >
        The Hidden Cost of Manual Workflows
      </h2>
      <div
        class="inline-block px-8 py-6 rounded-2xl mb-8"
        style="background: linear-gradient(135deg, var(--brand-navy) 0%, var(--brand-navy-light) 100%);"
      >
        <div
          class="text-center"
        >
          <h3
            class="text-4xl md:text-5xl lg:text-6xl font-inter font-bold text-white mb-2"
          >
            $236K - $330K
          </h3>
          <p
            class="text-xl md:text-2xl font-inter text-gray-200"
          >
            Annual Cost of Inefficiency
          </p>
        </div>
      </div>
    </div>
    <div
      class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-6 mb-16"
    >
      <div
        class="rounded-lg border bg-card text-card-foreground shadow-sm card-elevated hover:shadow-xl transition-all duration-300 hover:scale-105"
      >
        <div
          class="p-6 text-center"
        >
          <div
            class="w-16 h-16 mx-auto mb-4 bg-gradient-accent rounded-full flex items-center justify-center"
          >
            <div
              class="w-8 h-8 text-brand-navy"
              data-testid="clock-icon"
            />
          </div>
          <h3
            class="text-xl font-inter font-semibold text-brand-navy mb-3"
          >
            Time Drain
          </h3>
          <p
            class="text-secondary-600 leading-relaxed"
          >
            Manual processes consuming excessive employee hours and reducing productivity
          </p>
        </div>
      </div>
      <div
        class="rounded-lg border bg-card text-card-foreground shadow-sm card-elevated hover:shadow-xl transition-all duration-300 hover:scale-105"
      >
        <div
          class="p-6 text-center"
        >
          <div
            class="w-16 h-16 mx-auto mb-4 bg-gradient-accent rounded-full flex items-center justify-center"
          >
            <div
              class="w-8 h-8 text-brand-navy"
              data-testid="dollar-sign-icon"
            />
          </div>
          <h3
            class="text-xl font-inter font-semibold text-brand-navy mb-3"
          >
            Hidden Costs
          </h3>
          <p
            class="text-secondary-600 leading-relaxed"
          >
            Invisible expenses from inefficient workflows and repeated manual tasks
          </p>
        </div>
      </div>
      <div
        class="rounded-lg border bg-card text-card-foreground shadow-sm card-elevated hover:shadow-xl transition-all duration-300 hover:scale-105"
      >
        <div
          class="p-6 text-center"
        >
          <div
            class="w-16 h-16 mx-auto mb-4 bg-gradient-accent rounded-full flex items-center justify-center"
          >
            <div
              class="w-8 h-8 text-brand-navy"
              data-testid="alert-triangle-icon"
            />
          </div>
          <h3
            class="text-xl font-inter font-semibold text-brand-navy mb-3"
          >
            Error Prone
          </h3>
          <p
            class="text-secondary-600 leading-relaxed"
          >
            Human mistakes in manual processes leading to costly corrections and delays
          </p>
        </div>
      </div>
      <div
        class="rounded-lg border bg-card text-card-foreground shadow-sm card-elevated hover:shadow-xl transition-all duration-300 hover:scale-105"
      >
        <div
          class="p-6 text-center"
        >
          <div
            class="w-16 h-16 mx-auto mb-4 bg-gradient-accent rounded-full flex items-center justify-center"
          >
            <div
              class="w-8 h-8 text-brand-navy"
              data-testid="frown-icon"
            />
          </div>
          <h3
            class="text-xl font-inter font-semibold text-brand-navy mb-3"
          >
            Employee Burnout
          </h3>
          <p
            class="text-secondary-600 leading-relaxed"
          >
            Repetitive manual work reducing job satisfaction and increasing turnover
          </p>
        </div>
      </div>
    </div>
    <div
      class="text-center"
    >
      <div
        class="inline-flex items-center space-x-2 text-brand-navy font-inter font-semibold"
      >
        <span
          class="text-lg"
        >
          Ready to reclaim your time and reduce costs?
        </span>
        <div
          class="w-0 h-0 border-l-8 border-r-8 border-t-8 border-l-transparent border-r-transparent border-t-brand-navy animate-bounce"
        />
      </div>
      <p
        class="text-secondary-600 mt-2"
      >
        Discover how our intelligent automation solutions can transform your workflows below.
      </p>
    </div>
  </div>
</section>
`;
